protocol/messenger_handler.go

```diff
@@ -33,6 +34,11 @@ const (
 	transactionRequestDeclinedMessage           = "Transaction request declined"
 	requestAddressForTransactionAcceptedMessage = "Request address for transaction accepted"
 	requestAddressForTransactionDeclinedMessage = "Request address for transaction declined"
+
+	SyncWakuSectionKeyProfile     = "profile"
+	SyncWakuSectionKeyContacts    = "contacts"
+	SyncWakuSectionKeyCommunities = "communities"
+	SyncWakuSectionKeySettings    = "settings"
```

Comment:
Instead of defining the section keys here and later use them to hydrate `WakuSyncDetails`, how would you feel about encoding the possible sections in the struct itself?

So instead of having:

```golang
type SyncFromWakuResponse struct {
	SyncingProgressDetails map[string]protobuf.BackupDataDetails // key represents the data/section backup details refer to
	Profile                *SyncProfile
}
```

We'd have something like

```golang
type SyncingProgressDetails struct {
  Profile *protobuf.BackupDataDetails `json:"profile,omitempty"`
  Contacts *protobuf.BackupDataDetails  `json:"contacts,omitempty"`
  Communities *protobuf.BackupDataDetails  `json:"communities,omitempty"`
  Settings *protobuf.BackupDataDetails  `json:"settings,omitempty"`
}

type SyncFromWakuResponse struct {
	SyncingProgressDetails * SyncingProgressDetails
	Profile                *SyncProfile
}
```

That way it's explicit what data can be expected in a progress signal.
This would also allow to add a `TotalProgress` value in case this is interesting for clients.

I've done a similar thing for the discord import (look for `ImportProgress` if you need inspiration)

The downside is obviously that one can't add more backup details without changing the type as well.

Reply:
As you've said, it can be that way as well... what I had on my mind is that it will be a lot of sections/details for syncing, like more than 30 for sure, maybe more than 40 and having a map `map[string]protobuf.BackupDataDetails` looked more attractive to me, where key represents the section/detail we need to send a progress description for and value describes how many messages for that section we need to receive in total and which one current message refers to.

To summarize... this map `map[string]protobuf.BackupDataDetails` is a part of each backup message and it describes how many messages of which type we have backed up and which one is the received backup message in order.

Reply:
That's fair.

Can take this one offline but: how do you get to 30 or 40 things to be synced? I'd assume most of the stuff we're interested in lives already in profile, contacts, communities and settings

Reply:
> I'd assume most of the stuff we're interested in lives already in profile, contacts, communities and settings

Most likely you're referring to syncing among devices, that's true a lot of things are already in place for that syncing.

But in terms of backing up data to waku and fetching/handling them we had only contacts and communities, nothing else. I didn't updated that functionality just reorganize a code a bit so it fits new needs we have and add profile details (display name and identity images) to backup.

Here in this PR https://github.com/status-im/status-desktop/pull/8433 you can see how the changes applied here fit our need. 

When I said we will need to sync 30 - 40 things (will see when the list becomes final), yes we will have that much maybe even more but we will group them and display progress per group, for example as a part of profile syncing group will be some keycard related details, wallet details and so on.

But the point is that each backed up message carries complete information or part of information (depends if the thing we backed up is backed up as a single message or through more messages) and also the details of all backed up messages (how many of them we should receive for each thing and which one is the one being received in order).

Reply:
I think I am with @PascalPrecht on this one, basically we are talking about essentially the same data-struct, as a matter of fact it yields the same JSON, but one is using dynamic keys, while the other is using the struct fields.

A dynamic map use case is when you have dynamic keys, but here there's a set of well-defined keys.

The cost of change is pretty much identical, since you either add a new `const` or add a new field. Setting the map is probably easier, since you could have a generic function.

Though using fields seems more idiomatic and explicit to just use a struct, so I don't have to lookup somewhere else to understand what's in this struct, @Samyoul is the one that has likely the best eye for what's most idiomatic, but to me doesn't feel there's much advantage by using a dynamic map that's worth straying from what's more idiomatic